### PR TITLE
feat(skills): add resolve-pr-comments skill

### PR DESCRIPTION
**Pull request by scotthamilton77**
## Summary

Adds `resolve-pr-comments` skill at `src/user/.agents/skills/resolve-pr-comments/SKILL.md`. Companion to the existing `wait-for-pr-comments` skill:

- **`wait-for-pr-comments`** — polls for review arrival, auto-fixes the unambiguous slice
- **`resolve-pr-comments`** (new) — addresses what remains through a structured per-comment workflow

## Workflow

Six phases orchestrated against open PR comments:

1. **Inventory** — pull every open thread (inline + top-level review summaries) via GraphQL with pagination; detect outdated threads
2. **Triage** — classify each comment; bias toward judgment, escalate only when truly ambiguous
3. **Per-comment subagent dispatch** — one subagent per comment, `ralf-it` for significant work, full per-fix completion gate (code-reviewer + code-simplifier + verify-checklist)
4. **Final verification + push** — `verify-checklist` across combined work; handle non-fast-forward via pull-rebase (no force-push without auth)
5. **Reply + resolve** — reply to every comment (fixed AND skipped); resolve conversation via GraphQL for FIXED only
6. **Final report** — structured summary with explicit merge prohibition

## Development Process

Followed `superpowers:writing-skills` TDD methodology:

- **RED** — baseline subagent run without the skill showed batched commits, refusal to resolve human threads, missing final verify-checklist
- **GREEN** — wrote skill; compliance subagent followed per-comment dispatch, resolved FIXED only, handled top-level summaries correctly
- **REFACTOR** — `superpowers:code-reviewer` surfaced 2 BLOCKERs (broken Quick-Reference mutation, undocumented reply endpoint) + 3 MAJORs (edge cases, per-fix verify-checklist, CSO description); all addressed. `code-simplifier` tightened 262→242 lines by merging Common Mistakes into Red Flags. Final compliance test against 6-comment scenario (incl. outdated thread, top-level summary, non-fast-forward) passed flawlessly.

Tracked in bead `agents-config-mui`.

## Test plan
- [ ] Compliance verified via subagent against 6-comment scenario (passed)
- [ ] Author peer review
- [ ] Copilot automated review
- [ ] Document lint (render graphviz blocks)

### CODE REVIEW

File: src/user/.agents/skills/resolve-pr-comments/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+```dot
+digraph when_to_use {
+    "PR has open review comments?" [shape=diamond];
+    "Already auto-fixed by wait-for-pr-comments?" [shape=diamond];
+    "Done — nothing to address" [shape=box];
+    "Use this skill" [shape=doublecircle];
+    "Use wait-for-pr-comments first" [shape=box];
+
+    "PR has open review comments?" -> "Already auto-fixed by wait-for-pr-comments?" [label="yes"];
+    "PR has open review comments?" -> "Use wait-for-pr-comments first" [label="no — still polling"];
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply src/user/.agents/skills/resolve-pr-comments/SKILL.md]: Fixed in cdf9c9c: added "PR still being polled for first review?" diamond so the "no open comments" branch routes to either "Done — nothing to address" or "Use wait-for-pr-comments first" based on review state. Matches the logic in your suggestion.

File: src/user/.agents/skills/resolve-pr-comments/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+              nodes{ databaseId body path line author{login} }
+            }
+          }
+        }
+        reviews(first:50){
+          nodes{ id state body author{login} submittedAt }
+        }
+      }
+    }
+  }' -F owner=<owner> -F repo=<repo> -F number=<number>
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply src/user/.agents/skills/resolve-pr-comments/SKILL.md]: Fixed in cdf9c9c: added `$after: String` variable, wired `after:$after` into `reviewThreads(first:100, after:$after)`, and added `-F after=<cursor-or-empty>` so callers can actually paginate.

File: src/user/.agents/skills/resolve-pr-comments/SKILL.md [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+    repository(owner:$owner,name:$repo){
+      pullRequest(number:$number){
+        reviewThreads(first:100){
+          pageInfo{ hasNextPage endCursor }
+          nodes{
+            id isResolved isOutdated
+            comments(first:10){
+              nodes{ databaseId body path line author{login} }
+            }
+          }
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply src/user/.agents/skills/resolve-pr-comments/SKILL.md:91]: Fixed in cdf9c9c: bumped `comments(first:10)` to `first:100`, added `pageInfo{hasNextPage endCursor}` inside the inner comments selection, and documented per-thread comment pagination as a separate bullet.

File: src/user/.agents/skills/resolve-pr-comments/SKILL.md [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+| "New comments arrived mid-flight — I'll handle them next time" | Re-inventory after Phase 3; address in the same run. |
+| "Reply with 'Done.' — short and sweet" | Each reply must explain what changed (+ SHA) or why skipped. |
+| "Inline `<id>` directly into the GraphQL mutation" | Use `-F id=<thread_id>` with `mutation($id:ID!)` — variable binding only. |
+
+## Related Skills
+
+- **`wait-for-pr-comments`** — runs FIRST, polls and auto-fixes the unambiguous slice. This skill picks up what's left.
+- **`ralf-it`** — invoked INSIDE per-comment subagents for significant work; owns its own planning and refinement.
+- **`verify-checklist`** — per-fix (inside subagents) AND once at the end (orchestrator).
+- **`receiving-code-review`** — methodology for evaluating feedback (verify before implementing). Useful background.
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply src/user/.agents/skills/resolve-pr-comments/SKILL.md]: Fixed in cdf9c9c: removed the `receiving-code-review` entry from Related Skills. You're right — it's a superpowers-plugin skill that isn't a file in this repo. The remaining Related Skills entries (`wait-for-pr-comments`, `ralf-it`, `verify-checklist`) all have in-repo files.

File: .beads/issues.jsonl [hunk 5]
```diff
@@ -0,0 +1,17 @@
+{"id":"agents-config-mui","title":"Create resolve-pr-comments skill","description":"Add a dedicated skill that addresses PR review comments using a structured per-comment subagent workflow. Triggers: user wants to resolve open PR review comments (Copilot, human, or both). Workflow: clarify ambiguity only when judgment cannot, dispatch one subagent per fix, use ralf-it for significant work, run completion gate (code-review + code-simplifier + verify-checklist), commit back to PR, reply to each comment with what was fixed/skipped + reason, GH-resolve conversations only for FIXED comments. Lives in src/user/.agents/skills/resolve-pr-comments/SKILL.md (shared content, copied to all tools).","status":"in_progress","priority":2,"issue_type":"feature","assignee":"Scott Hamilton","owner":"scotthamilton77@gmail.com","created_at":"2026-04-19T12:05:23Z","created_by":"Scott Hamilton","updated_at":"2026-04-19T12:05:39Z","started_at":"2026-04-19T12:05:38Z","dependency_count":0,"dependent_count":0,"comment_count":0}
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply .beads/issues.jsonl]: Not addressed in this PR — needs project-policy decision first. The `scotthamilton77@gmail.com` in the `owner` field is the repo owner's public GitHub email (already visible on the GitHub profile and in commit authorship), and the `.beads/issues.jsonl` format is generated by the beads plugin with that field baked in. Redacting would mean either (a) changing the beads export schema upstream, or (b) gitignoring `.beads/issues.jsonl` entirely and relying on the Dolt sync path. Leaving this thread open for @scotthamilton77 to decide policy before I touch it.

[the remark this answers is not in the document]
Comment (scotthamilton77) [reply .beads/issues.jsonl]: Fixed in 022a094: added `issues.jsonl` to `.beads/.gitignore` and `git rm --cached`'d the file. Beads state now syncs via Dolt only for this repo. **Note:** prior commit 1972f6e still contains the file in PR history — forward-only fix; history scrub would require force-push not authorized in this session.

File: src/user/.agents/skills/resolve-pr-comments/SKILL.md [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+            id isResolved isOutdated
+            comments(first:100){
+              pageInfo{ hasNextPage endCursor }
+              nodes{ databaseId body path line author{login} }
+            }
+          }
+        }
+        reviews(first:50){
+          nodes{ id state body author{login} submittedAt }
+        }
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply src/user/.agents/skills/resolve-pr-comments/SKILL.md]: Fixed in 1359e6b: bumped `reviews(first:50)` to `first:100`, added `pageInfo{hasNextPage endCursor}` on the reviews connection, introduced `$reviewsAfter: String` variable alongside `$threadsAfter`, and documented looping both connections until `hasNextPage` is false.

File: src/user/.agents/skills/resolve-pr-comments/SKILL.md [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+  }' -F owner=<owner> -F repo=<repo> -F number=<number> -F after=<cursor-or-empty>
+```
+
+- **Top-level review summaries** live on `reviews.nodes` (non-empty `body`), NOT on `reviewThreads`.
+- **Thread pagination:** loop while `reviewThreads.pageInfo.hasNextPage` is true, passing `endCursor` as the next `after`. Don't silently truncate.
+- **Per-thread comment pagination:** `comments(first:100)` covers almost every real thread; if `comments.pageInfo.hasNextPage` is true for any thread, paginate that thread separately with the same pattern — you need the latest `databaseId` to reply.
+- **Outdated threads** (`isOutdated: true` after rebase): low-priority — reply acknowledging, resolve only if the underlying concern was actually addressed.
+
+Build a working list: `{ thread_id, comment_id, author, location, body, isResolved, isOutdated }`. Drop already-resolved.
+
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply src/user/.agents/skills/resolve-pr-comments/SKILL.md]: Fixed in 1359e6b: restructured the inventory with a `kind` discriminator ("review_thread" | "review_summary" | "issue_comment") and per-kind ID fields: `thread_id` + `reply_to_comment_id` for review_thread, `review_id` for review_summary, `issue_comment_id` for issue_comment. Added a field matrix table, extended ID Vocabulary with all five ID types, and added a Red Flag entry against overloading one `comment_id` field.

File: src/user/.agents/skills/resolve-pr-comments/SKILL.md [hunk 8]
```diff
@@ -0,0 +1,247 @@
...
+      thread{ isResolved }
+    }
+  }' -F id=<thread_id>
+```
+
+**FIXED — top-level review summary** (lives on `reviews.nodes`, no thread):
+```bash
+gh pr comment <number> --body "Addressed in <sha>: <summary>."
+# No resolve — top-level reviews have no resolvable thread.
+```
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply src/user/.agents/skills/resolve-pr-comments/SKILL.md]: Fixed in 1359e6b: Phase 5 now dispatches on `kind`. New sections for `review_summary` (gh pr comment referencing the review) and `issue_comment` (gh pr comment with cross-reference URL to `#issuecomment-<id>`). Added a kind→endpoint→resolvable matrix table at the top of Phase 5 so the response path is unambiguous for every feedback source.
